policies/policies.go

```diff
@@ -0,0 +1,301 @@
+	if err != nil {
+		logger.Errorf("lookupEffectivePolicies error: %v", err)
+		return
+	}
+
+	// We don't check the error from ospackage.SetConfig as all errors are already logged.
+	setConfig(resp)
+}
+
+// Run looks up osconfigs and applies them using tasker.Enqueue.
+func Run(ctx context.Context, res string) {
+	tasker.Enqueue("Run OSPackage", func() { run(ctx, res) })
+}
+
+func lookupEffectivePolicies(ctx context.Context, client *osconfig.Client, instance string) (*osconfigpb.LookupEffectiveGuestPoliciesResponse, error) {
+	var shortName, version, arch string
+	if config.OSInventoryEnabled() {
+		logger.Debugf("OS Inventory enabled for instance, gathering DistributionInfo for LookupEffectiveGuestPoliciesRequest")
+		info, err := osinfo.GetDistributionInfo()
+		if err != nil {
+			return nil, err
+		}
+		shortName = info.ShortName
+		version = info.Version
+		arch = info.Architecture
+	} else {
+		logger.Debugf("OS Inventory not enabled for instance, not gathering DistributionInfo for LookupEffectiveGuestPoliciesRequest")
+	}
+
+	identityToken, err := metadata.Get(config.IdentityTokenPath)
+	if err != nil {
+		return nil, err
+	}
+
+	req := &osconfigpb.LookupEffectiveGuestPoliciesRequest{
+		Instance:        instance,
+		OsShortName:     shortName,
+		OsVersion:       version,
+		OsArchitecture:  arch,
+		InstanceIdToken: identityToken,
+	}
+	logger.Debugf("LookupEffectiveGuestPolicies request: {Instance: %s, OsShortName: %s, OsVersion: %s, OsArchitecture: %s}",
+		req.GetInstance(), req.GetOsShortName(), req.GetOsVersion(), req.GetOsArchitecture())
+
+	res, err := client.LookupEffectiveGuestPolicies(ctx, req)
+	if err != nil {
+		if s, ok := status.FromError(err); ok {
+			return nil, fmt.Errorf("code: %q, message: %q, details: %q", s.Code(), s.Message(), s.Details())
+		}
+		return nil, err
+	}
+	logger.Debugf("LookupEffectiveGuestPolicies response:\n%s\n\n", dump.Sprint(res))
+
+	return res, nil
+}
+
+func setConfig(res *osconfigpb.LookupEffectiveGuestPoliciesResponse) error {
+	var aptRepos []*osconfigpb.AptRepository
+	var yumRepos []*osconfigpb.YumRepository
+	var zypperRepos []*osconfigpb.ZypperRepository
+	var gooRepos []*osconfigpb.GooRepository
+	for _, repo := range res.GetPackageRepositories() {
+		if r := repo.GetPackageRepository().GetGoo(); r != nil {
+			gooRepos = append(gooRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetApt(); r != nil {
+			aptRepos = append(aptRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetYum(); r != nil {
+			yumRepos = append(yumRepos, r)
+			continue
+		}
+		if r := repo.GetPackageRepository().GetZypper(); r != nil {
+			zypperRepos = append(zypperRepos, r)
+			continue
+		}
+	}
+
+	var aptInstalled, aptRemoved, aptUpdated []*osconfigpb.Package
+	var yumInstalled, yumRemoved, yumUpdated []*osconfigpb.Package
+	var zypperInstalled, zypperRemoved, zypperUpdated []*osconfigpb.Package
+	var gooInstalled, gooRemoved, gooUpdated []*osconfigpb.Package
```

Comment:
minor: Since this is desired state, consider names like "aptToInstall". This will help contrast these values with the maps of what is actually already installed or available for update.